composer/datasets/glue.py

```diff
@@ -95,6 +96,9 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
             raise MissingConditionalImportError(extra_deps_group="nlp", conda_package="transformers") from e
 
         self.validate()
+        assert self.task is not None, "checked in validate()"
+        assert self.tokenizer_name is not None, "checked in validate()"
```

Comment:
these checks are already happening in `validate`. If this is necessary for autoyahp + temporary, I'm good with it. Otherwise can we choose one or the other?

Reply:
> Definitely not `2`, a few of us have already run afoul on this while trying to debug runs on cpu. I don't think we can go wrong with 1, 3 or 4 as long as it's clear what is happening. My personal order or preference is 3. -> 4.  -> 1.  

Discussed offline; decided to go with option 4 (leave as is), as option 3 would require the yamls to specify `precision: amp`, which would defeat the portability.

Reply:
These additional checks are required since pyright cannot assume anything what `validate()` does.

But YAHP 0.1.1 will deprecate `validate()`. Instead, all parsing should be done in `initialize_object`. (This is why: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)